src/components/Editable.tsx

```diff
@@ -593,6 +593,7 @@ const Editable = ({
                 replaceEnd: selection.offsetEnd()!,
               }
             : null),
+          caretPosition: (selection.isText() ? selection.offset() || 0 : state.cursorOffset) || 0,
```

Comment:
Please provide a comment here, explaining why this is needed. Anything that is not obvious from the code should be explained in a comment. Here, we should note that `state.cursorOffset` may not be up-to-date.